src/ucp/core/ucp_mm.c

```diff
@@ -686,7 +687,107 @@ void ucp_mpool_obj_init(ucs_mpool_t *mp, void *obj, void *chunk)
 {
     ucp_mem_desc_t *elem_hdr  = obj;
     ucp_mem_desc_t *chunk_hdr = (ucp_mem_desc_t*)((ucp_mem_desc_t*)chunk - 1);
-    elem_hdr->memh = chunk_hdr->memh;
+    ucp_rndv_mpool_priv_t *mpriv;
+
+    mpriv              = (ucp_rndv_mpool_priv_t *)ucs_mpool_priv(mp);
+    elem_hdr->memh     = chunk_hdr->memh;
+    elem_hdr->ptr      = (void*)((ucp_mem_desc_t *)obj + 1);
+    elem_hdr->mem_type = mpriv->mem_type;
+}
+
+static inline ucs_status_t
```

Comment:
why inline?

Reply:
Following prototype for ucp_mpool_malloc function (Why was this made inline?). Should I make it non-inline?

Reply:
should be non-inline